Sources/Utility/ArgumentParser.swift

```diff
@@ -329,22 +308,17 @@ public final class ArgumentParser {
 
         /// Adds a result.
         ///
-        /// - parameter
+        /// - Parameters:
+        ///     - values: The associated values of the argument.
         ///     - argument: The argument for which this result is being added.
-        ///     - value: The associated value of the argument.
-        ///
-        /// - throws: ArgumentParserError
-        fileprivate func addResult(for argument: AnyArgument, result: ArgumentKind) throws {
+        fileprivate func add(_ values: [ArgumentKind], for argument: AnyArgument) throws {
             if argument.isArray {
-                var array = [ArgumentKind]()
-                // Get the previously added results if present.
-                if let previousResult = results[argument] as? [ArgumentKind] {
-                    array = previousResult
-                }
-                array.append(result)
+                var array = results[argument] as? [ArgumentKind] ?? []
+                array.append(contentsOf: values)
                 results[argument] = array
             } else {
-                results[argument] = result
+                assert(values.count == 1)
+                results[argument] = values.first!
```

Comment:
I initially chose a different path, but doing it this way really allows many simplifications in the parsing code. Basically, if the parser is modelled so that it always parses between `0` and `n` values for each argument, you can avoid a lot of `switch` and `if` statements that un-necessarily complicate the code. As long as a client of `ArgumentParser` still sees single values or arrays, I think it's a net win.

Comments?